TerrorConsole/Assets/_Main/Scripts/UI/Settings/AutoScrollDropdownItem.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+using DG.Tweening;
+
+namespace TerrorConsole
+{
+    public class AutoScrollDropdownItem : MonoBehaviour, ISelectHandler
+    {
+        [SerializeField] RectTransform _myRectTransform;
+        [SerializeField] Toggle _myToggle;
+        [SerializeField] ScrollRect _scrollRect;
+        [SerializeField] RectTransform contentPanel;
+
+        private void Start()
+        {
+            if(_myToggle.isOn)
+                UpdateScrollPosition();
+        }
+
+        public void OnSelect(BaseEventData eventData)
+        {
+            UpdateScrollPosition();
+        }
+
+        private void UpdateScrollPosition()
+        {
+            Vector2 targetPosition = (Vector2)_scrollRect.transform.InverseTransformPoint(contentPanel.position) - (Vector2)_scrollRect.transform.InverseTransformPoint(transform.position);
+
+            targetPosition -= new Vector2(0, _myRectTransform.rect.height);
+            contentPanel.DOAnchorPos(targetPosition, 0.3f).SetEase(Ease.OutCubic);
```

Comment:
Adjusting Scroll Normalized position did not work?

Reply:
`_scrollRect.content.childCount` to get the number of options.
`transform.GetSiblingIndex()` to now what gameObject index this script is attached to.

Your solution is creative but is far more complicated compared with what we can achieve with existing tools.

Save your code, only use it if the `_scrollRect.normalizePosition` route looks horrible when using.